### PR TITLE
Fix test suite on windows: skip on missing extensions, correct path flags

### PR DESCRIPTION
**Pull request by Tobion**
extracted from #718

### CODE REVIEW

File: tests/Doctrine/Tests/DBAL/Functional/ExceptionTest.php [hunk 1]
```diff
@@ -314,7 +315,8 @@ public function testConnectionExceptionSqLite($mode, $exceptionClass)
     public function getSqLiteOpenConnection()
     {
         return array(
-            array(0000, '\Doctrine\DBAL\Exception\ConnectionException'),
+            // mode 0 is considered read-only on Windows
+            array(0000, '\\' !== DIRECTORY_SEPARATOR ? '\Doctrine\DBAL\Exception\ConnectionException' : '\Doctrine\DBAL\Exception\ReadOnlyException'),
```
Comment (DHager) [inline tests/Doctrine/Tests/DBAL/Functional/ExceptionTest.php]: I'd rather not infer the host-OS based on the directory-separator. Maybe look at the `PHP_OS` constant instead? I've seen other projects check the presence of the `PHP_WINDOWS_VERSION_BUILD` constant as well.

Comment (deeky666) [inline tests/Doctrine/Tests/DBAL/Functional/ExceptionTest.php]: +1 we should check it with PHP constants.

Comment (Tobion) [inline tests/Doctrine/Tests/DBAL/Functional/ExceptionTest.php]: changed

**Comment by Tobion** [comment]
@deeky666 ping

**Comment by deeky666** [comment]
@Tobion thanks!

**Pull request merged**